base/shared/src/main/scala/scalaz/core/eq.scala

```diff
@@ -1,14 +1,16 @@
 package scalaz
 package core
 
+import scala.AnyRef
 import scala.language.experimental.macros
 
 trait EqClass[A] {
-  def equal(first: A, second: A): Boolean
+  def equal(first: A, second: A): Boolean = first == second
```

Comment:
I'd rather make a separate `EqAnyVal` at that point and keep `EqClass` completely abstract

Reply:
Well then that means, unfortunately, the specialized methods will never be used unless you have a reference to a statically typed `EqAnyVal`.

Reply:
oh yeah... :/

Reply:
I think that this means that this PR won't be nearly as good for performance as we hope.

Reply:
I am open to suggestions :)
It seems it would be easy enough to make this performant but I fear that usability would really suck. I am currently unsure how to make things fast **and** usable.

Reply:
How does `@specialized` on `EqClass` affect usability?

Reply:
I think I just misunderstood :)